eputnam/latrine
# Sidemenu

#### What this includes:

Primarily this PR includes an added popout sidemenu. The sidemenu will open when a restroom map marker is clicked. The sidemenu currently displays the Name, comments, and address for the corresponding restroom. The issue (https://github.com/eputnam/latrine/issues/21) describes having a photo and a phone number but these are not available in provided API and will need to be fetched from a difference resource.
- [ ] TODO: Make the side menu look prettier
---
### Screenshot
![image](https://user-images.githubusercontent.com/249293/42140660-2d2f987e-7d57-11e8-9695-c4b7ba1c918d.png)
---
### Details: 
In order to achieve this, I had to add a couple dependencies and upgrade react
Added dependencies:
- material-ui [https://github.com/mui-org/material-ui](https://github.com/mui-org/material-ui)
- styled-components [https://github.com/styled-components/styled-components](https://github.com/styled-components/styled-components)

I also created a `components` directory and moved a couple of the component files in there for maintainability.

I also moved the fetch for restrooms into a utility and is called from within the app component to pass the fetched data down to both the map and the sidemenu. As the app grows, we may consider some kind of state management library. Though the new context api may be able to provide this. I still need to learn more about that.

---
To test:
1. Pull this branch
2. `cd` indo the `frontend` directory
3. `npm install` to install new dependencies (sometimes if it breaks after installing new dependencies, you need to delete your node_modules directory and install again)
4.  npm run start to run the development server
5. Visit the port provided
6. Once the map markers have loaded, click on one
7. You should see a sidemenu appear from the left side with information about the restroom you clicked
